tests/slate/test_slate_infrastructure.py

```diff
@@ -66,15 +66,21 @@ def load(function_space):
 @pytest.fixture
 def boundary_load(function_space):
     f = Function(function_space)
-    f.interpolate(Expression("cos(x[1]*pi*2)"))
+    if function_space.mesh().cell_dimension == 1:
+        f.interpolate(Expression("cos(x[0]*pi*2)"))
+    else:
+        f.interpolate(Expression("cos(x[1]*pi*2)"))
```

Comment:
The reason that this generated illegal code rather than producing a nice error is that you are using the obsolete and unsafe C string interpolation. Please do this the nice way using Python objects. See: https://www.firedrakeproject.org/interpolation.html

Reply:
Done.